src/core/namespace.c

```diff
@@ -1905,7 +1907,10 @@ int setup_namespace(
         if (mount_flags == 0)
                 mount_flags = MS_SHARED;
 
-        if (root_image) {
+        if (root_image_unresolved) {
+                r = resolve_auto_dir(root_image_unresolved, false, &root_image);
```

Comment:
please add a comment `/* want_dir= */` before the `false` to make clear what the argument is about. `false` alone is otherwise so opaque